gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/monitor/PollingConfigurationAnalyzer.java

```diff
@@ -365,12 +375,32 @@ private DiscoveryApiClient getApiClient(final ServiceDiscoveryConfig discoveryCo
                                                clusterName,
                                                lastTimestamp);
     for (ApiEvent event : events) {
-      restartEvents.add(new RestartEvent(event));
+      List<ApiEventAttribute> attributes = event.getAttributes();
+      Map<String,Object> map = getAttributeMap(attributes);
```

Comment:
Why not loop through the ApiEventAttribute list directly? There doesn't seem to be any value in converting to a Map. You break the map apart in the same way you would have to interrogate the ApiEventAttribute anyway.

Reply:
That is what I had before and it is much more complicated as I wanted to avoid creating a map. Note that we are checking two correlated attributes. The command issued and its status. Since you don't know what order they are in you have to squirrel them away then check whether you have both then add. If we end up having to add more events that would be even messier to do. Now, we can get the map upfront and do whatever processing of it afterward much more cleanly.

Reply:
Hmm I guess I am now more concerned with the logic here potentially then. I'll comment on the specific section below.